### ISSUE TITLE
TIMELINE: Event labels wrap on FF 3 on Mac

### ISSUE THREAD

**Issue by GoogleCodeExporter** [berusch/simile-widgets-old#13]
```
Firefox 3 on Mac does not calculate the label widths correctly. Causes
wrapping of (some!) longer labels.

 Note this is FF3 Mac specific. Works fine on Google Chrome, IE 7, FF
3-win, Safari on Mac and Win.

Workaround and more details in the group emails.
(Need to provide ref urls) 

Test case
View the Cubism example on FF 3 on Mac
```

Original issue reported on code.google.com by `larryklu...@gmail.com` on 19 Sep 2008 at 3:52

**Comment by GoogleCodeExporter** [comment]
```
David's work-around overrides 
SimileAjax.Graphics._FontRenderingContext.prototype.computeSize and fixes the 
symptom but not the cause. Details here:

http://groups.google.com/group/simile-widgets/msg/63a3239771fdb5ac?dmode=source

```

Original comment by `crumpj...@gmail.com` on 21 Sep 2008 at 6:00

**Comment by GoogleCodeExporter** [comment]
```
I spent some time on this problem today and narrowed things down to a specific 
problem of bad values being returned from the offsetWidth property in certain 
circumstances on Firefox/Mac. I built a test page:
http://simile-
widgets.googlecode.com/svn/timeline/trunk/src/webapp/examples/test_example/firef
ox_ma
c_test_case.html

And filed an official bug with the Firefox folks:
https://bugzilla.mozilla.org/show_bug.cgi?id=458617

The offsetWidth property appears to be about 1% too small. Increasing it by the 
larger of [1% of the reported width, 1px]  solves the problem. Eg a short label 
of 
"Test 1" has a reported width of 28px on FF/Mac and 29px on Safari/Mac. The 
text 
wraps in FF. Increasing the width to 29px stops the wrapping.

I'll add the workaround (for FF/Mac only browsers) to the source sometime 
relatively 
soon. 

We'll see what the Firefox folks do with the bug report. If you want to be 
notified 
when the bug is changed, you can add yourself to the bug's cc list.

Regards,

Larry
```

Original comment by `larryklu...@gmail.com` on 5 Oct 2008 at 7:38
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `stefano.mazzocchi@gmail.com` on 25 Mar 2009 at 7:01
- Added labels: **Project-Timeline**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 2.3.0
```

Original comment by `larryklu...@gmail.com` on 30 Mar 2009 at 7:58
- Changed state: **Fixed**
- Added labels: **Fixed-in-2.3.0**
